apps/src/templates/hintsDisplay.jsx

```diff
@@ -0,0 +1,57 @@
+module.exports = React.createClass({
+
+  propTypes: {
+    hintReviewTitle: React.PropTypes.string.isRequired,
+    getSeenHints: React.PropTypes.func.isRequired,
+    getUnseenHints: React.PropTypes.func.isRequired,
+    recordUserViewedHint: React.PropTypes.func.isRequired,
+    lightbulbSVG: React.PropTypes.node.isRequired,
+  },
+
+  getInitialState: function () {
+    return {
+      seenHints: this.props.getSeenHints(),
+      unseenHints: this.props.getUnseenHints(),
+      showViewHintButton: true
+    };
+  },
+
+  viewHint: function () {
+    this.props.recordUserViewedHint(this.state.unseenHints[0]);
+    var newState = this.getInitialState();
```

Comment:
I think that the way that I would structure this is as follows:

props consist of seen hints, unseen hints, and a function recordViewedHint

state consist only of showViewHintButton

when i click view hint, it calls recordUserViewedHint, which I'm assuming updates which hints we have/haven't seen. it then calls React.Render again, which causes us to rerender the component using the new props